bigquery_etl/util.py

```diff
@@ -142,13 +190,33 @@ def load_views(bq: bigquery.Client, dataset: bigquery.Dataset, views: Dict[str,
 
 def read(*paths: str, decoder: Optional[Callable] = None, **kwargs):
     """Read a file and apply decoder if provided."""
-    with open(os.path.join(*paths), **kwargs) as f:
-        return decoder(f) if decoder else f.read()
+    filepath = os.path.join(*paths)
+    with open(filepath, **kwargs) as f:
+        return decoder(f, filepath) if decoder else f.read()
 
 
-def ndjson_load(file_obj: Iterable[str]) -> List[Any]:
+def ndjson_load(file_obj: Iterable[str], filepath: str) -> List[Any]:
     """Decode newline delimited json from file_obj."""
-    return [json.loads(line) for line in file_obj]
+    res = []
+    for i, line in enumerate(file_obj):
+        try:
+            res.append(json.loads(line))
+        except json.JSONDecodeError as e:
+            raise NDJsonDecodeError(
+                f"Line {i+1} column {e.colno} of file {filepath}, {e.msg}"
+            )
+
+    return res
+
+
+def json_load(file_obj: Iterable[str], filepath: str) -> List[Any]:
+    """Decode json from file_obj."""
+    try:
+        return json.loads("".join(file_obj))
+    except json.JSONDecodeError as e:
+        raise JsonDecodeError(
+            f"Line {e.lineno} column {e.colno} of file {filepath}, {e.msg}"
+        )
```

Comment:
These should use `file_obj.name`, rather than passing it in redundantly, like so:
```python
from io import TextIOWrapper


def read(*paths: str, decoder: Optional[Callable] = None, **kwargs):
    """Read a file and apply decoder if provided."""
    with open(os.path.join(*paths), **kwargs) as f:
        return decoder(f) if decoder else f.read()


def ndjson_load(file_obj: TextIOWrapper) -> List[Any]:
    """Decode newline delimited json from file_obj."""
    try:
        return [json.loads(line) for line in file_obj]
    except json.JSONDecodeError as e:
        raise NDJsonDecodeError(
            f"Line {i+1} column {e.colno} of file {file_obj.name}, {e.msg}"
        )


def json_load(file_obj: TextIOWrapper) -> List[Any]:
    """Decode json from file_obj."""
    try:
        return json.load(file_obj)
    except json.JSONDecodeError as e:
        raise JsonDecodeError(
            f"Line {e.lineno} column {e.colno} of file {file_obj.name}, {e.msg}"
        )
```